connectors/sources/mysql.py

```diff
@@ -569,6 +573,14 @@ async def _yield_docs_custom_query(self, client, tables, query):
         )
         column_names = await client.get_column_names_for_query(query=query)
 
+        if not set(primary_key_columns) - set(column_names):
+            pass
+        else:
+            self._logger.warning(
+                f"Skipping query {query} for tables {', '.join(tables)} as primary key column name/id_column is not present in query."
+            )
+            return
```

Comment:
```suggestion
        if set(primary_key_columns) - set(column_names):
            self._logger.warning(
                f"Skipping query {query} for tables {', '.join(tables)} as primary key column name/id_column is not present in query."
            )
            return
```

this feels less convoluted.